baseapp-chats/baseapp_chats/graphql/filters.py

```diff
@@ -1,36 +1,55 @@
 import django_filters
 import swapper
 from baseapp_core.graphql import get_pk_from_relay_id
-from django.db.models import Q
+from django.db.models import OuterRef, Q, Subquery
 
 ChatRoom = swapper.load_model("baseapp_chats", "ChatRoom")
+ChatRoomParticipant = swapper.load_model("baseapp_chats", "ChatRoomParticipant")
 
 
 class ChatRoomFilter(django_filters.FilterSet):
     q = django_filters.CharFilter(method="filter_q")
     profile_id = django_filters.CharFilter(method="filter_profile_id")
     unread_messages = django_filters.BooleanFilter(method="filter_unread_messages")
+    archived = django_filters.BooleanFilter(method="filter_archived")
 
     order_by = django_filters.OrderingFilter(fields=(("created", "created"),))
 
     class Meta:
         model = ChatRoom
         fields = ["q", "order_by"]
 
+    @property
+    def qs(self):
+        qs = super().qs
+        archived = self.data.get("archived", False)
+        try:
+            user_profile = self.request.user.current_profile
+        except AttributeError:
+            return qs.none()
+
+        non_archived_rooms_subquery = ChatRoomParticipant.objects.filter(
+            room=OuterRef("pk")
+        ).filter(profile=user_profile, has_archived_room=archived)
```

Comment:
the filter for archive should be on `filter_archived`, and instead of doing a subquery, maybe you could do: `qs.filter(participants__profile_id=user.profile.pk, participants__has_archived_room=archived)`

and can you please add a test for this new case?